### PR TITLE
enable common nfs handler to deal with nfs direct mount

### PR DESCRIPTION
**Pull request by liangjingtao11**
1. Match NFS PVC and obtain NFS source path based on the mount information on the host
2. Create a target directory and perform a mount operation in guest using create container hook
3. supports subpath mount and readonly mount

### CODE REVIEW

File: pkg/plugins/vmvol/nfs_csi.go [hunk 1]
```diff
@@ -1,53 +1,69 @@
...
-
 	"github.com/containerd/nri/pkg/api"
 	"github.com/kcrow-io/kcrow/pkg/vmvol"
+	linux "github.com/moby/sys/mountinfo"
+	"k8s.io/klog/v2"
 )
 
 func init() {
-	vmvol.RegistVolHandler("nfs.csi", nfsCsiHandler)
+	vmvol.RegistVolHandler("nfs.csi", commonHandler)
```
Comment (yylt) [inline pkg/plugins/vmvol/nfs_csi.go:11]: name 'common' ?

File: pkg/vmvol/manage.go [hunk 2]
```diff
@@ -78,17 +77,20 @@ func (m *manager) Process(ctx context.Context, im *oci.Item) error {
...
 			}
-			// for kata runtime, the destination has prefix
-			// /run/kata-containers/$cid/rootfs
+
 			podvols = append(podvols, &PodVol{
-				Destination: path.Join(kataPrefixPath(im.Ct), volmnt.MountPath),
+				VolumeMount: volmnt,
+				Container:   im.Ct,
 				PvSpec:      v,
+				GuestPath:   path.Join(kataPrefixPath(im.Ct), volmnt.MountPath),
```
Comment (yylt) [inline pkg/vmvol/manage.go:91]: Destination may be good

File: pkg/vmvol/manage.go [hunk 3]
```diff
@@ -78,17 +77,20 @@ func (m *manager) Process(ctx context.Context, im *oci.Item) error {
...
 			continue
 		}
+
 		for _, volmnt := range cnt.VolumeMounts {
 			v, ok := vols[volmnt.Name]
 			if !ok {
 				continue
 			}
-			// for kata runtime, the destination has prefix
-			// /run/kata-containers/$cid/rootfs
```
Comment (yylt) [inline pkg/vmvol/manage.go:87]: 保留注释

File: pkg/plugins/vmvol/nfs_csi.go [hunk 4]
```diff
@@ -6,48 +6,76 @@ import (
...
+				for _, info := range nfsMountInfo {
+					if info.Mountpoint == m.Source {
+						args = append(args, info.Source)
+						nfsFound = true
+						break
+					}
+				}
+
+				if nfsFound {
+					destination := path.Join(kataPrefixPath(podv.Container), m.Destination)
```
[the remark this answers is not in the document]
Comment (liangjingtao11) [reply pkg/plugins/vmvol/nfs_csi.go]: done，现在已经回退[pkg/vmvol/manage.go 目录

File: pkg/vmvol/manage.go [hunk 5]
```diff
@@ -104,13 +103,9 @@ func (m *manager) Process(ctx context.Context, im *oci.Item) error {
...
 				}
-				im.Adjust.RemoveMount(strings.TrimLeft(res.Destination, kataPrefixPath(im.Ct)))
+				im.Adjust.RemoveMount(res.Destination)
 			}
 		}
 	}
 	return nil
 }
-
-func kataPrefixPath(cnt *api.Container) string {
```
[the remark this answers is not in the document]
Comment (liangjingtao11) [reply pkg/vmvol/manage.go]: done

File: pkg/plugins/vmvol/nfs_csi.go (hunk 1 above)
Comment (liangjingtao11) [reply pkg/plugins/vmvol/nfs_csi.go]: ok，nfs.common看起来更合适。

File: pkg/vmvol/manage.go (hunk 2 above)
Comment (liangjingtao11) [reply pkg/vmvol/manage.go]: done

File: pkg/vmvol/manage.go (hunk 3 above)
Comment (liangjingtao11) [reply pkg/vmvol/manage.go:87]: done

**Review by yylt — APPROVED** [review]

**Pull request merged**